src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/OpenAmDAO.scala

```diff
@@ -0,0 +1,101 @@
+package org.broadinstitute.dsde.workbench.sam.openam
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport.{sprayJsValueUnmarshaller => _, _}
+import akka.http.scaladsl.marshalling.{Marshal, Marshaller}
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.{Unmarshal, Unmarshaller}
+import akka.stream.Materializer
+import org.broadinstitute.dsde.workbench.sam.WorkbenchExceptionWithErrorReport
+import org.broadinstitute.dsde.workbench.sam.directory.{DirectoryConfig, DirectorySubjectNameSupport}
+import org.broadinstitute.dsde.workbench.sam.model.OpenAmJsonSupport._
+import org.broadinstitute.dsde.workbench.sam.model._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Created by mbemis on 5/22/17.
+  */
+class OpenAmDAO(openAmConfig: OpenAmConfig, protected val directoryConfig: DirectoryConfig)(implicit val system: ActorSystem, val materializer: Materializer, val executionContext: ExecutionContext) extends DirectorySubjectNameSupport {
+
+  private def authorizationHeader(userInfo: UserInfo) = headers.RawHeader("iPlanetDirectoryPro", userInfo.accessToken)
+  private val jsonContentTypeHeader = headers.`Content-Type`(ContentTypes.`application/json`)
+
+  def getAdminUserInfo: Future[UserInfo] = getAuthToken(openAmConfig.user, openAmConfig.password).map(response => UserInfo(response.tokenId, SamUserId(openAmConfig.user)))
+
+  def getAuthToken(username: String, password: String): Future[AuthenticateResponse] = {
+    val authenticationUrl = openAmConfig.url + "/json/authenticate"
+    val usernameHeader = headers.RawHeader("X-OpenAM-Username", username)
+    val passwordHeader = headers.RawHeader("X-OpenAM-Password", password)
+
+    httpRequest[String, AuthenticateResponse](authenticationUrl, Option("{}"), HttpMethods.POST, List(usernameHeader, passwordHeader))
+  }
+
+  def listResourceTypes(userInfo: UserInfo): Future[Set[OpenAmResourceType]] = {
+    val resourceTypesUrl = openAmConfig.url + "/json/resourcetypes?_queryFilter=true"
+
+    httpRequest[String, OpenAmResourceTypeList](resourceTypesUrl, None, HttpMethods.GET, List(authorizationHeader(userInfo))).map(_.result)
```

Comment:
really I want to do something like...

`httpRequest[_, OpenAmResourceTypeList]...`, since a GET isn't marshalling any payload